### PR TITLE
[build] Fix JDK8 builds and set up CI

### PR DESCRIPTION
**Pull request by nisargthakkar**
## Fix JDK8 builds and set up CI

Previously, builds using JDK8 would fail. Also, cleaned up the `alpn-boot` loading process for HTTP/2 on JDK 8 tests.

## How was this PR tested?

Running GH CI

## Does this PR introduce any user-facing changes?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: .github/rawWorkflows/gh-ci-alpini-parametrized-flow.txt
Comment (FelixGV) [inline .github/rawWorkflows/gh-ci-alpini-parametrized-flow.txt:31]: Microsoft is not available for jdk8?

**Review by FelixGV — APPROVED** [review]
Nice, thanks for the cleanups!

File: .github/rawWorkflows/gh-ci-alpini-parametrized-flow.txt
Comment (nisargthakkar) [reply .github/rawWorkflows/gh-ci-alpini-parametrized-flow.txt:31]: Yeah. Microsoft's docs suggest to use `temurin` instead: https://learn.microsoft.com/en-us/java/openjdk/download#openjdk-8. We could split the matrix to use `microsoft` for jdk 11 and 17, and `temurin` for jdk 8 if we wish to

Comment (FelixGV) [reply .github/rawWorkflows/gh-ci-alpini-parametrized-flow.txt:31]: Hmm... either way is fine I guess. I can give another ship it if you prefer changing it. Your call.

Comment (nisargthakkar) [reply .github/rawWorkflows/gh-ci-alpini-parametrized-flow.txt:31]: I think we can keep it this way and change it later if there are problems

**Pull request merged**
